pkg/appgateserver/server.go

```diff
@@ -123,8 +123,9 @@ func NewAppGateServer(
 
 	// TODO_CONSIDERATION: Use app.listeningEndpoint scheme to determine which
 	// kind of server to create (HTTP, HTTPS, TCP, UNIX, etc...)
-	// TODO_BLOCKER(red0ne): Use TCP server for the AppGateServer to have generic support
-	// of RPCTypes.
+	// TODO_RESEARCH: Currently, the communication between the AppGateServer and the
```

Comment:
@studna FYI, these are captured in TODO comments.

Related to this discussion:

![Screenshot 2024-06-04 at 2 07 03 PM](https://github.com/pokt-network/poktroll/assets/1892194/e0d1433d-cc55-43dd-bed9-13ba0bf6cebf)